src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/StrategyDraggableItem/StrategyItem/StrategyExecution/StrategyExecution.tsx

```diff
@@ -241,43 +248,65 @@ export const StrategyExecution = ({ strategy }: IStrategyExecutionProps) => {
             }
             return null;
         });
-    };
+    }, [parameters, definition, strategy, constraints, styles]);
 
-    return (
-        <>
-            <ConditionallyRender
-                condition={Boolean(uiConfig.flags.SE)}
-                show={<FeatureOverviewSegment strategyId={strategy.id} />}
-            />
-            <ConditionallyRender
-                condition={constraints.length > 0}
-                show={
-                    <>
-                        <ConstraintAccordionList
-                            constraints={constraints}
-                            showLabel={false}
-                        />
-                        <StrategySeparator text="AND" />
-                    </>
-                }
-            />
-            <ConditionallyRender
-                condition={strategy.name === 'default'}
-                show={
-                    <Box sx={{ width: '100%' }} className={styles.summary}>
-                        The standard strategy is{' '}
-                        <Chip
-                            variant="outlined"
-                            size="small"
-                            color="success"
-                            label="ON"
-                        />{' '}
-                        for all users.
-                    </Box>
-                }
+    if (!parameters) {
+        return <NoItems />;
+    }
+
+    const listItems = [
+        Boolean(uiConfig.flags.SE) && segments && segments.length > 0 && (
+            <FeatureOverviewSegment strategyId={strategy.id} />
+        ),
+        constraints.length > 0 && (
+            <ConstraintAccordionList
+                constraints={constraints}
+                showLabel={false}
             />
-            {renderParameters()}
-            {renderCustomStrategy()}
-        </>
+        ),
+        strategy.name === 'default' && (
+            <>
+                <ConditionallyRender
+                    condition={constraints.length > 0 || Boolean(segments)}
+                    show={<StrategySeparator text="AND" />}
+                />
+                <Box sx={{ width: '100%' }} className={styles.summary}>
+                    The standard strategy is{' '}
+                    <Chip
+                        variant="outlined"
+                        size="small"
+                        color="success"
+                        label="ON"
+                    />{' '}
+                    for all users.
+                </Box>
+            </>
+        ),
+        Boolean(renderParameters) &&
+            (renderParameters as any[])?.length > 0 &&
+            renderParameters,
+        Boolean(renderCustomStrategy) &&
+            (renderCustomStrategy as any[])?.length > 0 &&
+            renderCustomStrategy,
+    ].filter(Boolean);
```

Comment:
If any part of the list is empty, this will remove it to display dividers between in a correct way.